### ISSUE TITLE
File/Folder Permissions with Groups

### ISSUE THREAD

**Issue by evandam** [jupyterhub/jupyterhub#968]
I am running into an issue where Jupyterhub doesn't seem to be able to recognize permissions through AD/Vintela groups. Since these groups are not local, they do not appear in /etc/group, so I'm not sure if that's what Jupyterhub attempts to read from to determine permissions.

I'm running Jupyterhub from RHEL 6.8, and reproducing the issue with the following:
```
mkdir test
chmod 700 test
# Can view the directory in Jupyterhub without a problem up to this point.
# Force Jupyterhub to read group-level permissions. pcusers is a local group.
chown <another_user>:pcusers test 
chmod 070 test
# Can still view the directory since it's using a local group.
chgrp ad-enabled-group test
# Can no longer view the folder in Jupyterhub using an AD/Vintela group.
```
I have confirmed that I am in the groups the correct groups, and I can navigate to them through the command line.

Any help is appreciated!

**Comment by willingc** [comment]
Hi @evandam,

I'm not familiar with how AD/Vintela works for permissions/authentication. Since as you say the groups are not local, I suspect that some additional customization or configuration may be needed. You may be able to use something similar to the [shared notebook service examples](https://github.com/jupyterhub/jupyterhub/blob/master/examples/service-notebook/README.md) in this repo.

@minrk may have some additional insights to this.

**Comment by evandam** [comment]
Thanks for the quick reply! As far as I can tell, the fact that they are AD/Vintela groups is transparent and they all still appear with the `groups` command.
The shared notebook seems worth looking into, but I would like users to be able to explore any files they should have access to if that makes sense.

**Comment by willingc** [comment]
> I would like users to be able to explore any files they should have access to if that makes sense.

@evandam Can you clarify what you mean here? Thanks.

**Comment by evandam** [comment]
Hi @willingc,

Outside of a Jupyter Notebook (in the file explorer view, ex: `localhost:8000/user/<user>/tree`), I'm trying to navigate to different directories. I have the below config set that lands a user in their home directory by default, but they can navigate all the way to the top of the file system.
```
c.Spawner.notebook_dir = '/'
c.Spawner.default_url = '/tree/home/%U'
```
When I click the "home" icon now, it brings me to the root of the file system, which is what I'd expect with the config above. The goal isn't necessarily to allow users to share notebooks, but more to allow users to browse the entire file system as needed to view other files. However, the issue with groups seems to be what prevents this from working.

Hopefully that helps!

**Comment by willingc** [comment]
Hi @evandam,

Thanks for the explanation. Let's see if we can get this working for you. I would suggest modifying the `default_url` to the one below. [Reference](https://jupyterhub.readthedocs.io/en/latest/troubleshooting.html#i-want-access-to-the-whole-filesystem-but-still-default-users-to-their-home-directory)

```
c.Spawner.notebook_dir = '/'
c.Spawner.default_url = '/home/%U' # %U will be replaced with the username
```

If that doesn't fully resolve it, feel free to continue asking questions here or on the gitter channel for jupyterhub.

**Comment by evandam** [comment]
Hi @willingc,

That's what I had been using initially, but I get a 404 error when I sign in and click "My Server," which attempts to go to `localhost:8000/user/<USER>/home/<USER>`. I found that adding /tree resolved the issue, though.

**Comment by willingc** [comment]
Hi @evandam, If you post a debug log with the error that would help us troubleshoot this. Thanks!

**Comment by evandam** [comment]
Hi @willingc 

Here's the log after I try running Jupyterhub with the `notebook_dir` and `default_url` set as above:
```
[I 2017-02-09 11:56:56.784 JupyterHub app:622] Loading cookie_secret from /usr/app/jupyter/cookie_secret
[W 2017-02-09 11:56:56.819 JupyterHub app:304]
    Generating CONFIGPROXY_AUTH_TOKEN. Restarting the Hub will require restarting the proxy.
    Set CONFIGPROXY_AUTH_TOKEN env or JupyterHub.proxy_auth_token config to avoid this message.

[I 2017-02-09 11:56:56.828 JupyterHub app:785] Not using whitelist. Any authenticated user will be allowed.
[I 2017-02-09 11:56:56.848 JupyterHub app:1231] Hub API listening on http://127.0.0.1:8081/hub/
[W 2017-02-09 11:56:56.853 JupyterHub app:959] Running JupyterHub without SSL. There better be SSL termination happening somewhere else...
[I 2017-02-09 11:56:56.853 JupyterHub app:968] Starting proxy @ http://*:8000/
11:56:57.065 - info: [ConfigProxy] Proxying http://*:8000 to http://127.0.0.1:8081
11:56:57.069 - info: [ConfigProxy] Proxy API at http://127.0.0.1:8001/api/routes
[I 2017-02-09 11:56:57.159 JupyterHub app:1254] JupyterHub is now running at http://127.0.0.1:8000/
11:57:01.206 - debug: [ConfigProxy] PROXY WEB / to http://127.0.0.1:8081
[I 2017-02-09 11:57:01.227 JupyterHub log:100] 302 GET / (@10.147.194.181) 1.70ms
11:57:01.309 - debug: [ConfigProxy] PROXY WEB / to http://127.0.0.1:8081
[I 2017-02-09 11:57:01.316 JupyterHub log:100] 302 GET / (@10.147.194.181) 0.81ms
11:57:01.360 - debug: [ConfigProxy] PROXY WEB /hub to http://127.0.0.1:8081
[I 2017-02-09 11:57:01.363 JupyterHub log:100] 302 GET /hub (@10.147.194.181) 0.47ms
11:57:01.440 - debug: [ConfigProxy] PROXY WEB /hub/ to http://127.0.0.1:8081
[I 2017-02-09 11:57:01.451 JupyterHub log:100] 302 GET /hub/ (ev80173@10.147.194.181) 3.29ms
11:57:01.494 - debug: [ConfigProxy] PROXY WEB /hub/home to http://127.0.0.1:8081
[I 2017-02-09 11:57:01.543 JupyterHub log:100] 200 GET /hub/home (ev80173@10.147.194.181) 42.26ms
11:57:03.591 - debug: [ConfigProxy] PROXY WEB /hub/spawn to http://127.0.0.1:8081
[I 2017-02-09 11:57:03.603 JupyterHub log:100] 302 GET /hub/spawn (ev80173@10.147.194.181) 4.54ms
11:57:03.699 - debug: [ConfigProxy] PROXY WEB /user/ev80173 to http://127.0.0.1:8081
[I 2017-02-09 11:57:03.703 JupyterHub log:100] 302 GET /user/ev80173 (@10.147.194.181) 0.80ms
11:57:03.826 - debug: [ConfigProxy] PROXY WEB /hub/user/ev80173 to http://127.0.0.1:8081
[I 2017-02-09 11:57:03.911 JupyterHub spawner:467] Spawning /usr/app/anaconda_3/bin/jupyterhub-singleuser --user=ev80173 --port=51803 --cookie-name=jupyter-hub-token-ev80173 --base-url=/user/ev80173 --hub-host= --hub-prefix=/hub/ --hub-api-url=http://127.0.0.1:8081/hub/api --ip=127.0.0.1 --notebook-dir=/ --NotebookApp.default_url=/home/ev80173
[I 2017-02-09 11:57:06.814 ev80173 manager:32] [nb_conda_kernels] enabled, 2 kernels found
[I 2017-02-09 11:57:07.533 ev80173 handlers:290] [nb_conda] enabled
[I 2017-02-09 11:57:07.621 ev80173 __init__:35] ✓ nbpresent HTML export ENABLED
[W 2017-02-09 11:57:07.622 ev80173 __init__:43] ✗ nbpresent PDF export DISABLED: No module named 'nbbrowserpdf'
[I 2017-02-09 11:57:07.675 ev80173 handlers:73] [nb_anacondacloud] enabled
[I 2017-02-09 11:57:07.683 ev80173 notebookapp:1128] Serving notebooks from local directory: /
[I 2017-02-09 11:57:07.683 ev80173 notebookapp:1128] 0 active kernels
[I 2017-02-09 11:57:07.683 ev80173 notebookapp:1128] The Jupyter Notebook is running at: http://127.0.0.1:51803/user/ev80173/
[I 2017-02-09 11:57:07.683 ev80173 notebookapp:1129] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[I 2017-02-09 11:57:07.686 ev80173 log:47] 302 GET /user/ev80173 (127.0.0.1) 0.92ms
[I 2017-02-09 11:57:07.687 JupyterHub base:306] User ev80173 server took 3.857 seconds to start
[I 2017-02-09 11:57:07.687 JupyterHub orm:159] Adding user ev80173 to proxy /user/ev80173 => http://127.0.0.1:51803
11:57:07.693 - debug: [ConfigProxy] POST /user/ev80173 target=http://127.0.0.1:51803, user=ev80173
[I 2017-02-09 11:57:07.705 JupyterHub log:100] 302 GET /hub/user/ev80173 (ev80173@10.147.194.181) 3872.76ms
11:57:07.770 - debug: [ConfigProxy] PROXY WEB /user/ev80173 to http://127.0.0.1:51803
[I 2017-02-09 11:57:07.774 ev80173 log:47] 302 GET /user/ev80173 (10.147.194.181) 1.22ms
11:57:07.952 - debug: [ConfigProxy] PROXY WEB /user/ev80173/home/ev80173 to http://127.0.0.1:51803
[I 2017-02-09 11:57:07.982 JupyterHub log:100] 200 GET /hub/api/authorizations/cookie/jupyter-hub-token-ev80173/[secret] (ev80173@127.0.0.1) 13.55ms
[W 2017-02-09 11:57:08.013 ev80173 log:47] 404 GET /user/ev80173/home/ev80173 (10.147.194.181) 59.28ms referer=http://lad1labhc2089:8000/hub/home
11:57:08.154 - debug: [ConfigProxy] PROXY WEB /user/ev80173/custom/custom.css to http://127.0.0.1:51803

```

**Comment by willingc** [comment]
@evandam Thanks for the additional debug output.

@minrk When you have time will you take a look at this as I'm not sure if this is a configuration issue or would be resolved by #925. Thanks!

**Comment by willingc** [comment]
Closing this issue since it hasn't been updated in a while. If it the issue still exists, please leave a comment requesting this issue be reopened. Thanks! 🌻
